src/main/java/frc/robot/subsystems/Launcher.java

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkBase.ControlType;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class Launcher extends SubsystemBase {
+  /** Creates a new Launcher. */
+  private CANSparkMax upper =
+      new CANSparkMax(Constants.LaunchConstants.upperCANID, MotorType.kBrushless);
+
+  private CANSparkMax lower =
+      new CANSparkMax(Constants.LaunchConstants.lowerCANID, MotorType.kBrushless);
+
+  private boolean tuningPIDS = false;
+
+  private DoubleSolenoid angleSwitcher =
+      new DoubleSolenoid(
+          PneumaticsModuleType.CTREPCM,
+          Constants.LaunchConstants.angleSwitchForwardChannel,
+          Constants.LaunchConstants.angleSwitchReverseChannel);
+
+  public Launcher() {
+    setPIDsDefault();
+    showPIDs();
+  }
+
+  private void showPIDs() {
+    SmartDashboard.putNumber("Launch P", upper.getPIDController().getP());
+    SmartDashboard.putNumber("Launch I", upper.getPIDController().getI());
+    SmartDashboard.putNumber("Launch D", upper.getPIDController().getD());
+  }
+
+  private void setPIDsDefault() {
+    upper.getPIDController().setP(Constants.LaunchConstants.launcherP);
+    upper.getPIDController().setI(Constants.LaunchConstants.launcherI);
+    upper.getPIDController().setD(Constants.LaunchConstants.launcherD);
+
+    lower.getPIDController().setP(Constants.LaunchConstants.launcherP);
+    lower.getPIDController().setI(Constants.LaunchConstants.launcherI);
+    lower.getPIDController().setD(Constants.LaunchConstants.launcherD);
+  }
+
+  private void updatePIDs(double p, double i, double d) {
+    upper.getPIDController().setP(p);
+    upper.getPIDController().setI(i);
+    upper.getPIDController().setD(d);
+
+    lower.getPIDController().setP(p);
+    lower.getPIDController().setI(i);
+    lower.getPIDController().setD(d);
+  }
+
+  private void updatePIDFromDashboard() {
+    double p = SmartDashboard.getNumber("Launch P", 0);
+    double i = SmartDashboard.getNumber("Launch I", 0);
+    double d = SmartDashboard.getNumber("Launch D", 0);
```

Comment:
i tthink youu need to call updatePIDs(p, i, d)

and you could make this all in one line:

```java
updatePIDS(
    SmartDashboard.getNumber("Launch P", 0),
    SmartDashboard.getNumber("Launch I", 0),
    SmartDashboard.getNumber("Launch D", 0))
```